draft-ietf-webtrans-http3.md

```diff
@@ -166,18 +172,90 @@ The `webtransport` HTTP Upgrade Token uses the Capsule Protocol as defined in
 
 ## Limiting the Number of Simultaneous Sessions
 
-From the flow control perspective, WebTransport sessions count against the
-stream flow control just like regular HTTP requests, since they are established
-via an HTTP CONNECT request.  This document does not make any effort to
-introduce a separate flow control mechanism for sessions, nor to separate HTTP
-requests from WebTransport data streams.  If the server needs to limit the rate
-of incoming requests, it has alternative mechanisms at its disposal:
+This document defines a SETTINGS_MAX_WEBTRANSPORT_SESSIONS parameter that allows
+the server to limit the maximum number of concurrent WebTransport sessions on a
+single HTTP/3 connection.  The client MUST NOT open more sessions than
+indicated in the server SETTINGS parameters.  The server MUST NOT close the
+connection if the client opens sessions exceeding this limit, as the client and
+the server do not have a consistent view of how many sessions are open due to
+the asynchronous nature of the protocol; instead, it MUST reset all of the
+CONNECT streams it is not willing to process with the `HTTP_REQUEST_REJECTED`
+status defined in [HTTP3].
+
+### WT_MAX_STREAMS Capsule {#WT_MAX_STREAMS}
+
+*[WT_MAX_STREAMS]: #
+
+An HTTP capsule {{HTTP-DATAGRAM}} called WT_MAX_STREAMS is introduced to inform
+the peer of the cumulative number of streams of a given type it is permitted to
+open.  A WT_MAX_STREAMS capsule with a type of 0x12 applies to bidirectional
+streams, and a WT_MAX_STREAMS capsule with a type of 0x13 applies to
+unidirectional streams.
+
+~~~
+WT_MAX_STREAMS Capsule {
+  Type (i) = 0x12..0x13,
```

Comment:
Similarly, let's use capsule types in the 4-byte encoding while we're still iterating